File: BankingAppFrontend/src/app/account.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Account } from './account';

@Injectable({
  providedIn: 'root'
})
export class AccountService {
  
  private baseUrl = 'http://localhost:8080/api/accounts';

  constructor(private httpClient: HttpClient) {}

  getAllAccounts(): Observable<Account[]> {
    return this.httpClient.get<Account[]>(this.baseUrl);
  }

  

  createAccount(account:Account):Observable<Account>{
    return this.httpClient.post<Account> (`${this.baseUrl}`,account)
  }
  getAccountById( id:number):Observable<Account>{
    return this.httpClient.get<Account>(`${this.baseUrl}/${id}`)
  }
  deleteAccount(id: number): Observable<{ message: string }> {  
    return this.httpClient.delete<{ message: string }>(`${this.baseUrl}/${id}`);
  }
  
  
  
  deposit(accountId: number, amount: number): Observable<Account> {
    return this.httpClient.put<Account>(`${this.baseUrl}/${accountId}/deposit`, { amount });
  }

  withdraw(accountId: number, amount: number): Observable<Account> {
    return this.httpClient.put<Account>(`${this.baseUrl}/${accountId}/withdraw`, { amount });
  }
  
}

File: BankingAppBackend/src/main/java/com/BankingApp/service/AccountService.java
package com.BankingApp.service;

import java.util.List;

import com.BankingApp.dto.AccountDto;

public interface AccountService {
    AccountDto createAccount(AccountDto accountDto);
    AccountDto getAccountByID(Long  id); 


    
    
    AccountDto deposit(Long id,double amount);
    
    AccountDto withdraw(Long id, double amount);
    
    List <AccountDto>getAllAccounts();
    boolean deleteAccount(Long id);
    
    
}
